Wikipedia/Code/DiffContainerViewController.swift

```diff
@@ -0,0 +1,495 @@
+        let previousRevision = revisionDelegate?.retrievePreviousRevision(with: fromModel)
+        
+        let singleDiffVC = DiffContainerViewController(articleTitle: articleTitle, siteURL: siteURL, type: .single(byteDifference: fromModel.revisionSize), fromModel: previousRevision, toModel: fromModel, theme: theme, revisionDelegate: revisionDelegate)
+        wmf_push(singleDiffVC, animated: true)
+        
+    }
+    
+    @objc func tappedUp() {
+        
+        //note because we aren't filtering in History yet, PageHistoryViewController should always be able to tell us the next revision. If filtering is implemented this method will fail, and we will need to have DiffContainerViewController know how to handle a situation where fromModel is populated but toModel is not (that is, hide header & list, fetch next toModel & diff, then show header & list)
+        guard let nextRevision = revisionDelegate?.retrieveNextRevision(with: toModel) else {
+            assertionFailure("Unable to determine next revision. Perhaps user tapped the latest revision in history? Up arrow should be disabled in this case.")
+            return
+        }
+        
+        let singleDiffVC = DiffContainerViewController(articleTitle: articleTitle, siteURL: siteURL, type: .single(byteDifference: nextRevision.revisionSize), fromModel: toModel, toModel: nextRevision, theme: theme, revisionDelegate: revisionDelegate)
+        wmf_push(singleDiffVC, animated: true)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        switch type {
+        case .compare:
+            self.showDiffPanelOnce()
+        case .single:
+            break
+        }
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        if let scrollView = diffListViewController?.scrollView {
+            configureExtendedViewSquishing(scrollView: scrollView)
+        }
+
+        if let emptyViewController = scrollingEmptyViewController {
+            navigationBar.setNeedsLayout()
+            navigationBar.layoutSubviews()
+            let bottomSafeAreaHeight = view.bounds.height - safeAreaBottomAlignView.frame.maxY
+            let targetRect = CGRect(x: 0, y: navigationBar.visibleHeight, width: emptyViewController.view.frame.width, height: emptyViewController.view.frame.height - navigationBar.visibleHeight - bottomSafeAreaHeight)
+            //tonitodo: this still doesn't seem quite centered...
+            let convertedTargetRect = view.convert(targetRect, to: emptyViewController.view)
+            print(convertedTargetRect)
+            emptyViewController.centerEmptyView(within: convertedTargetRect)
+        }
+    }
+    
+    override func apply(theme: Theme) {
+        
+        guard isViewLoaded else {
+            return
+        }
+        
+        super.apply(theme: theme)
+        
```

Comment:
shouldn't super be called first so that `self.theme` can be set regardless of whether the view is loaded?

Reply:
good catch, fixed